src/include/kompute/TypeContainer.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ABCTypeContainer.hpp"
+#include <typeinfo>
+
+
+struct IdCounter
+{
+    static size_t counter;
+};
+
+template<typename T>
+class TypeContainer : public ABCTypeContainer, IdCounter
+{
+  private:
+    size_t classId()
+    {
+        static size_t id = counter++;
```

Comment:
This introduces non-deterministic behaviour, one run TypeContainer<float> will be different to another run - we should avoid. The typeid + std::typeinfo seemed more robust, what's the reason to not use here?

Reply:
changed to typeid